extensions/arc/src/models/controllerModel.ts

```diff
@@ -123,64 +124,68 @@ export class ControllerModel {
 		}
 		// create a new in progress promise object
 		ControllerModel._refreshInProgress = new Deferred<void>();
-		await this.azdataLogin(promptReconnect);
+		const session = await this.acquireAzdataLoginSession(promptReconnect);
 		const newRegistrations: Registration[] = [];
-		await Promise.all([
-			this._azdataApi.azdata.arc.dc.config.show().then(result => {
-				this._controllerConfig = result.result;
-				this.configLastUpdated = new Date();
-				this._onConfigUpdated.fire(this._controllerConfig);
-			}).catch(err => {
-				// If an error occurs show a message so the user knows something failed but still
-				// fire the event so callers can know to update (e.g. so dashboards don't show the
-				// loading icon forever)
-				if (showErrors) {
-					vscode.window.showErrorMessage(loc.fetchConfigFailed(this.info.name, err));
-				}
-				this._onConfigUpdated.fire(this._controllerConfig);
-				throw err;
-			}),
-			this._azdataApi.azdata.arc.dc.endpoint.list(this.azdataAdditionalEnvVars).then(result => {
-				this._endpoints = result.result;
-				this.endpointsLastUpdated = new Date();
-				this._onEndpointsUpdated.fire(this._endpoints);
-			}).catch(err => {
-				// If an error occurs show a message so the user knows something failed but still
-				// fire the event so callers can know to update (e.g. so dashboards don't show the
-				// loading icon forever)
-				if (showErrors) {
-					vscode.window.showErrorMessage(loc.fetchEndpointsFailed(this.info.name, err));
-				}
-				this._onEndpointsUpdated.fire(this._endpoints);
-				throw err;
-			}),
-			Promise.all([
-				this._azdataApi.azdata.arc.postgres.server.list(this.azdataAdditionalEnvVars).then(result => {
-					newRegistrations.push(...result.result.map(r => {
-						return {
-							instanceName: r.name,
-							state: r.state,
-							instanceType: ResourceType.postgresInstances
-						};
-					}));
+		try {
+			await Promise.all([
+				this._azdataApi.azdata.arc.dc.config.show(this.azdataAdditionalEnvVars, session).then(result => {
+					this._controllerConfig = result.result;
+					this.configLastUpdated = new Date();
+					this._onConfigUpdated.fire(this._controllerConfig);
+				}).catch(err => {
+					// If an error occurs show a message so the user knows something failed but still
+					// fire the event so callers can know to update (e.g. so dashboards don't show the
```

Comment:
I know the comment is not newly added, but I just noticed it doesn't make sense: 'can know to update'?

Reply:
It's not the clearest - but it means we fire the event so that anything hooking into the event know that *something* changed and should update accordingly. 

I'll see if I can reword it to be clearer.